app/models/site_channel_details.rb

```diff
@@ -109,4 +111,10 @@ def brave_publisher_id_not_changed_once_initialized
       errors.add(:brave_publisher_id, "can not change once initialized")
     end
   end
+
+  def same_publisher_brave_publisher_id_unique
+    if self.class.joins(:channel).where(brave_publisher_id: brave_publisher_id, "channels.publisher_id": channel.publisher_id).any?
```

Comment:
I believe `.where(brave_publisher_id: brave_publisher_id, "channels.publisher_id": channel.publisher_id)`

Can be re-written as `.where(brave_publisher_id: brave_publisher_id, channels: {publisher_id: channel.publisher_id})`